core/src/main/java/org/jobrunr/scheduling/ScheduleFactory.java

```diff
@@ -0,0 +1,24 @@
+package org.jobrunr.scheduling;
+
+import org.jobrunr.scheduling.cron.CronExpression;
+import org.jobrunr.scheduling.exceptions.InvalidScheduleException;
+import org.jobrunr.scheduling.interval.Interval;
+
+public class ScheduleFactory {
+
+    private ScheduleFactory() {}
+
+    public static Schedule getSchedule(String scheduleExpression){
+        ScheduleExpressionType type = ScheduleExpressionType.getScheduleType(scheduleExpression);
+
+        if(type.equals(ScheduleExpressionType.CRON_EXPRESSION)){
+            return CronExpression.create(scheduleExpression);
+        }
+        else if(type.equals(ScheduleExpressionType.INTERVAL)){
+            return new Interval(scheduleExpression);
+        }
+        else {
+            throw new InvalidScheduleException("Schedule expression cannot be mapped to any type");
+        }
+    }
+}
```

Comment:
Let's merge this with the `ScheduleExpressionType`.

The enum can also have a method `create` which returns then the correct `Schedule`.

Reply:
I moved it to the `ScheduleExpressionType` class but isn't it a little odd to have a method inside an enum return an instance of `Schedule`? Any thoughts?